src/test/uint256_tests.cpp

```diff
@@ -330,6 +333,59 @@ BOOST_AUTO_TEST_CASE(parse)
     }
 }
 
+/**
+ * Implemented as a templated function so it can be reused by other classes that have a FromHex()
+ * method that wraps base_blob::FromHex(), such as transaction_identifier::FromHex().
+ */
+template <typename T>
+void TestFromHex()
+{
+    constexpr unsigned int num_chars{T::size() * 2};
+    static_assert(num_chars <= 64); // this test needs to be modified to allow for more than 64 hex chars
+    const std::string valid_64char_input{"0123456789abcdef0123456789ABCDEF0123456789abcdef0123456789ABCDEF"};
+    const auto valid_input{valid_64char_input.substr(0, num_chars)};
+    {
+        // check that lower and upper case hex characters are accepted
+        auto valid_result{T::FromHex(valid_input)};
+        BOOST_REQUIRE(valid_result);
+        BOOST_CHECK_EQUAL(valid_result->ToString(), ToLower(valid_input));
+    }
+    {
+        // check that only strings of size num_chars are accepted
+        BOOST_CHECK(!T::FromHex(""));
+        BOOST_CHECK(!T::FromHex("0"));
+        BOOST_CHECK(!T::FromHex(valid_input.substr(0, num_chars / 2)));
+        BOOST_CHECK(!T::FromHex(valid_input.substr(0, num_chars - 1)));
+        BOOST_CHECK(!T::FromHex(valid_input + "0"));
+    }
+    {
+        // check that non-hex characters are not accepted
+        std::string invalid_chars{R"( !"#$%&'()*+,-./:;<=>?@GHIJKLMNOPQRSTUVWXYZ[\]^_`ghijklmnopqrstuvwxyz{|}~)"};
+        for (auto c : invalid_chars) {
+            BOOST_CHECK(!T::FromHex(valid_input.substr(0, num_chars - 1) + c));
+        }
+        // 0x prefixes are invalid
+        std::string invalid_prefix{"0x" + valid_input};
```

Comment:
this will fail because of the length requirerement, not the prefix, right?
```suggestion
        std::string invalid_prefix{"0x" + valid_input.substr(0, num_chars - 2)};
```

Reply:
Correct, this specific case is to ensure "0x" is not treated as some magic prefix that is silently trimmed. So I'm testing both the case of length 64 and 66 in the next 2 lines, so I'm going to keep this as is.